katas/content/grovers_search/index.md

```diff
@@ -0,0 +1,327 @@
+# Grover's Search Algorithm
+
+@[section]({
+    "id": "grovers_search__overview",
+    "title": "Overview"
+})
+
+This kata introduces you to Grover's search algorithm - one of the fundamental algorithms in quantum computing.
+
+**This kata covers the following topics:**
+
+- The general problem solved by Grover's search algorithm - the search problem
+- Implementing Grover's algorithm in Q# for a problem provided as a quantum oracle
+- Some of the practical aspects of this algorithm
+
+Note that this tutorial does not cover implementing specific classical functions as quantum oracles in detail. To get familiar with that topic, check out the earlier Oracles and Marking Oracles katas.
+
+**What you should know to start working on this kata:**
+
+- Basic knowledge of quantum gates and measurements
+- Basic understanding of quantum oracles
+
+
+@[section]({
+    "id": "grovers_search__search_problem",
+    "title": "The Search Problem"
+})
+
+The problem solved by Grover's search algorithm is called the *search problem* and can be formulated as follows.
+You are given a classical function that takes an $n$-bit input and returns a one-bit output $f(x): \{0, 1\}^n \to \{0, 1\}$. 
+The task is to find an input $x_0$ for which $f(x_0) = 1$.
+
+Importantly, you don't have any information about the internal structure of the function $f$! It is given to you as a "black box" that allows you to evaluate the value of the function for any output you want, but not to learn anything about how it works.
```

Comment:
for any output you want -> for any input you want?

Reply:
Nice catch!